### PR TITLE
[docs] consolidate all docs into sub folder

### PR DESCRIPTION
**Pull request by feministy**
👋 goodness, that was a lot of typing! this is the first step of many for #5026.

the only section that isn't complete with these changes is `RELEASING.md`, which is in progress in #5252 (those changes will be moved into the proper location if this PR goes in first).

i made a few minor tweaks to the project's README to support this organization. i also left the code of conduct in the root.

~~one thing i *am concerned about* is that our little "guidelines for contributing" warning message on PRs is going to disappear after this gets merged because `CONTRIBUTING.md` is no longer in the root~~

~~should i put a file in the root called `CONTRIBUTING.md` that links to the docs for contributing? 🤔  i'm not 100% sure on how the github integration for that works so y'all let me know.~~

### CODE REVIEW

**Comment by indirect** [comment]
Awesome! 😁 

I believe we can use either `docs/CONTRIBUTING.md` or `.github/CONTRIBUTING.md`. Needs more research.

**Comment by indirect** [comment]
OMG I looked through this and it is soooooo good. Now I am going to have to deal with what to do about the fact that I have been linking to https://github.com/bundler/bundler/blob/master/ISSUES.md for literally five years. Maybe I can make git.io shortlinks.

**Comment by segiddins** [comment]
CONTRIBUTING.md needs to stay at the root of the repo or in a .github folder to show up when people open PRs/issues

**Comment by feministy** [comment]
@segiddins i will move the overview document to the root 👌 

thanks @indirect!!

**Comment by feministy** [comment]
ok! the integration definitely requires it to be in the project root. i created a file `CONTRIBUTING.md` in the root, and used it as kind of a "quick start" guide to contributions with a link to the comprehensive documentation!

**Comment by indirect** [comment]
woo @bundlerbot r+

File: doc/development/NEW_FEATURES.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Adding New Features
+
+If you would like to add a new feature to Bundler, please follow these steps:
+
+  1. [Create an issue](https://github.com/bundler/bundler/issues/new) with the [`feature-request` label](https://github.com/bundler/bundler/labels/type:%20feature-request) to discuss your feature.
```
Comment (segiddins) [inline doc/development/NEW_FEATURES.md]: non-org members can't add labels

File: doc/development/PULL_REQUESTS.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+Make sure the code formatting and styling adheres to the guidelines. We use Rubocop for this. Lack of formatting adherence will result in automatic Travis build failures.
+
+      $ bin/rubocop -a
+
+## Tests
+
+Prior to submitting your PR, please run the test suite:
+
+      $ bin/rspec
```
Comment (segiddins) [inline doc/development/PULL_REQUESTS.md]: I'd prefer we advise using `bin/rake` (it guarantees the man pages have been built)

**Comment by segiddins** [comment]
@bundlerbot retry

**Pull request merged**